src/view/treeNodes/pullRequestNode.ts

```diff
@@ -903,9 +907,35 @@ export class PRNode extends TreeNode implements CommentHandler, vscode.Commentin
 
 	// #region Review
 	public async startReview(thread: GHPRCommentThread, input: string): Promise<void> {
-		await this._prManager.startReview(this.pullRequestModel);
-		await this.createOrReplyComment(thread, input);
-		this.setContextKey(true);
+		const temporaryCommentId = this.optimisticallyAddComment(thread, input, true);
+
+		try {
+
+			const fileChange = await this.findMatchingFileNode(thread.uri);
+			const position = this.calculateCommentPosition(fileChange, thread);
+			const newComment = await this._prManager.startReview(this.pullRequestModel,
+				{
+					body: input,
+					path: fileChange.fileName,
+					position
+				}
+			);
+
+			await this.updateCommentThreadCache(thread, fileChange, newComment);
+			this.replaceTemporaryComment(thread, newComment, temporaryCommentId);
+			fileChange.update(fileChange.comments.concat(newComment));
+			this.setContextKey(true);
+		} catch (e) {
+			vscode.window.showErrorMessage(`Starting a review failed: ${e}`);
+
+			thread.comments = thread.comments.map(c => {
```

Comment:
Consider using a for..of loop on the existing array.

Reply:
Ah, the assignment is actually necessary here - this triggers a rerender on the VSCode side. So I am going to leave as is. But thanks for taking the time to look at this! 😄